grails-app/controllers/org/pih/warehouse/invoice/InvoiceController.groovy

```diff
@@ -107,7 +108,7 @@ class InvoiceController {
             redirect(action: "list")
         }
 
-        Invoice invoice = invoiceService.generateInvoice(order)
+        Invoice invoice = prepaymentInvoiceService.generateInvoice(order)
```

Comment:
What method is getting called when generating an invoice for a PO that wasn't pre-paid? If this API is used in non-prepay related flows we don't want to have it use the prepaymentInvoiceService

Reply:
nvm I see the first line in the method checks that it has a prepayment invoice already.

Have you considered moving these methods to a PrepaymentInvoiceController since you're also doing that for the service methods?

Reply:
I think it's not worth it, to be honest. Later we will add PrepaymentInvoiceApiController, so this one will have only one endpoint.